doc/man7/password-handling.pod

```diff
@@ -0,0 +1,95 @@
+=pod
+
+=head1 NAME
+
+password handling - How diverse parts of OpenSSL treat pass phrases
+
+=head1 DESCRIPTION
+
+In a modern world with all sorts of character encodings, the treatment of pass
+phrases has become increasingly complex.
+This manual page attempts to give an overview over how this problem is
+currently addressed in different parts of the openssl library.
+
+=head2 The general case
+
+The OpenSSL library doesn't treat pass phrases in any special way as a general
+rule, and trusts the application or user to choose a suitable character set
+and stick to that throughout the life of affected objects.
+This means that for an object that was encrypted using a pass phrase encoded in
+ISO-8859-1, that object needs to be decrypted using a pass phrase encoded in
+ISO-8859-1.
+Anything else will have the decryption fail.
+
+=head2 PKCS#12
+
+PKCS#12 is a bit different regarding pass phrase encoding.
+The standard stipulates that the pass phrase shall be encoded as a BMPString,
+i.e. UTF-16.
+
+OpenSSL tries to adapt to this requirements in one of the following manners:
```

Comment:
Can  you please clarify where (which app, which functions) that try to do this?

Reply:
I'm not sure what the question is.  Is it sufficient if I say "anything that's related to PKCS#12 where pass phrases are used"?  That's *all* our documented PKCS#12 functions (`PKCS12_parse`, `PKCS12_create`, `PKCS12_newpass`).  That's also the command `openssl pkcs12`, as well as `openssl storeutl` and related functions (`OSSL_STORE_load`) when it's loading a PKCS#12 file.  Is that the answer you're looking for?